app/controllers/api/v1/search_results_controller.rb

```diff
@@ -3,37 +3,46 @@
 module Api
   module V1
     class SearchResultsController < ApplicationController
-      before_action :doorkeeper_authorize!
+      include Pagination
 
       def create
-        upload_form = UploadForm.new(
-          search_engine: upload_form_params[:search_engine],
-          csv_file: upload_form_params[:csv_file]
-        )
-
         if upload_form.valid?
           search_results = upload_form.save
-          render_success(search_results)
+
+          message = I18n.t('activemodel.notices.models.search_result.create')
+          search_results_serializer = SearchResultsSerializer.new(search_results, meta: { message: message })
+
+          render_success(serializer: search_results_serializer, status: :created)
         else
-          render_failed(upload_form)
+          render_error(detail: upload_form.errors.full_messages, status: :unprocessable_entity)
         end
       end
 
-      private
+      def index
+        search_result_list = current_user.search_results.order(:id)
+        pagy, paginated_results = paginated_resources_for(search_result_list)
 
-      def upload_form_params
-        params.permit(:search_engine, :csv_file)
+        search_result_serializer = SearchResultsSerializer.new(paginated_results, meta: meta_from_pagy(pagy))
+
+        render_success(serializer: search_result_serializer, status: :ok)
       end
 
-      def render_success(search_results)
-        data = SearchResultsSerializer.new(search_results).serializable_hash[:data]
-        message = I18n.t('activemodel.notices.models.search_result.create')
+      private
+
+      def upload_form
+        @upload_form ||= UploadForm.new(
+          search_engine: upload_form_params[:search_engine],
+          csv_file: upload_form_params[:csv_file],
+          user_id: current_user.id
+        )
+      end
 
-        render json: { data: data, meta: { message: message } }, status: :created
+      def upload_form_params
+        params.permit(:search_engine, :csv_file)
       end
 
-      def render_failed(upload_form)
-        render json: { errors: upload_form.errors.full_messages }, status: :unprocessable_entity
+      def render_success(serializer:, status:)
```

Comment:
Do we need this method as it's just one line?
If we intend to use it in other controllers, should we move it into the ApplicationController instead?